android/geowidget/src/main/java/org/smartregister/fhircore/geowidget/screens/GeoWidgetFragment.kt

```diff
@@ -311,33 +306,44 @@ class GeoWidgetFragment : Fragment() {
     )
   }
 
+  fun handleGeoJsonFeatures(geoJsonFeatures: List<GeoJsonFeature>) {
+    if (geoJsonFeatures.isNotEmpty()) {
+      geoWidgetViewModel.updateMapFeatures(geoJsonFeatures)
+      zoomMapWithFeatures()
+    }
+  }
+
   private fun zoomMapWithFeatures() {
     mapView?.getMapAsync { mapboxMap ->
-      val featureCollection =
-        FeatureCollection.fromFeatures(geoWidgetViewModel.mapFeatures.toList())
-      val locationPoints =
-        featureCollection
-          .features()
-          ?.asSequence()
-          ?.filter { it.geometry() is Point }
-          ?.map { it.geometry() as Point }
-          ?.toMutableList() ?: emptyList()
-
-      val bbox = TurfMeasurement.bbox(MultiPoint.fromLngLats(locationPoints))
-      val paddedBbox = CoordinateUtils.getPaddedBbox(bbox, 1000.0)
-      val bounds = LatLngBounds.from(paddedBbox[3], paddedBbox[2], paddedBbox[1], paddedBbox[0])
-      val finalCameraPosition = CameraUpdateFactory.newLatLngBounds(bounds, 50)
-
-      with(mapboxMap) {
-        (style?.getSourceAs(requireContext().getString(R.string.data_set_quest)) as GeoJsonSource?)
-          ?.apply { setGeoJson(featureCollection) }
-        easeCamera(finalCameraPosition)
+      val features = geoWidgetViewModel.mapFeatures.toList()
+      if (features.isNotEmpty()) {
+        val featureCollection = FeatureCollection.fromFeatures(features)
+        val locationPoints =
+          featureCollection
+            .features()
+            ?.asSequence()
+            ?.filter { it.geometry() is Point }
+            ?.map { it.geometry() as Point }
+            ?.toMutableList() ?: emptyList()
+
+        val bbox = TurfMeasurement.bbox(MultiPoint.fromLngLats(locationPoints))
+        val paddedBbox = CoordinateUtils.getPaddedBbox(bbox, 1000.0)
+        val bounds = LatLngBounds.from(paddedBbox[3], paddedBbox[2], paddedBbox[1], paddedBbox[0])
```

Comment:
I don't think so, but does kotlin have a destructure into args syntax? I think in Python we could do *paddedBox[:3].reverse() instead of adding all the indices

Reply:
Yea, refactoring this to a more Kotlinish code will make this more readable. Lemme check. This implementation borrowed from Kujaku Java code.

Reply:
Kotlin has a reversed function.

Reply:
Oh! we can't update this to use List, the function is overloaded, and the values are function params.